server-common/src/main/java/org/apache/cassandra/sidecar/common/server/CQLSessionProvider.java

```diff
@@ -31,23 +31,24 @@
 public interface CQLSessionProvider
 {
     /**
-     * Provides a Session connected to the cluster. If null it means the connection was
-     * could not be established. The session still might throw a NoHostAvailableException if the
-     * cluster is otherwise unreachable.
+     * Provides a Session connected to the cluster; otherwise, it tries to connect to the cluster.
+     * Returning null means the connection can not be established.
```

Comment:
Why null? Would it we worth to throw a different exception? Something like `ConnectionCouldNotBeStablishedException`?

Reply:
The API has been like this that can return nullable result. I am only updating the JavaDoc here to clarify the behavior. Any API change here is unrelated and should be avoided. 

I like the idea of throwing exception instead of returning null in this case. It seems to be able to improve the code clarity. Let's give it more thought. And probably do it in a follow-up.